### ISSUE TITLE
Prettier configuration should not place embedded components on newlines

### ISSUE THREAD

**Issue by alfarok** [w3f/polkadot-wiki#3704]
Currently, running `npx pretty-quick` may reformat a paragraph placing `<RPC>` elements on their own line due to the longer nature.

For example:

![image](https://user-images.githubusercontent.com/13341935/186714324-22909cb1-bc0e-4a41-93a5-7d4eebdcc7c7.png)

The issue stems from the markdown parser expecting a newline to start with a character, if it doesn't it assumes end of paragraph and adds a bottom margin to that paragraph:

![image](https://user-images.githubusercontent.com/13341935/186715761-9a3cf7f9-d971-47dd-ba79-1dded0bf7e07.png)

Since the formatting comes from the parent component, not the RPC component, I am hesitant to mess with any higher-level CSS. A small change that works here could incorrectly format other content on the site.  Chunks of markdown can be ignore from the prettier action if specifically tagged, such as:

![image](https://user-images.githubusercontent.com/13341935/186714583-77040643-bc7f-4328-b9da-71504808d724.png)

But this is redundant and not an ideal solution.  This tasks should involve finding a better solution that will not force developers to have to double check for reformatted RPC components after running the command.

A couple possible routes are:
- Can a rule be added to the `.prettierrc.json` file to ignore HTML elements or something similar
- Is the {{ polkadot: :polkadot }} conditional rendering syntax also impacted by this?  If not can RPC components be wrapped in `{{ }}` to prevent formatting etc.

**Comment by alfarok** [comment]
After spending more time today looking into various solutions for this, I think using
```
<!-- prettier-ignore-start -->
```
```
<!-- prettier-ignore-end -->
```

comment syntax is the best solution for now.

If after running prettier any in-line react components become improperly re-formatted, undo the prettier action in visual studio, wrap the block between the 2 listed comments above, re-run prettier.  The formatter will now ignore everything between the tags.

Not the best solution but if we were to ensure every document that contains an RPC component, is correctly formatted after running prettier (adding the comment tag if required), we would just need to be cognizant when adding new instances.  This way we would have confidence running the command, knowing it wouldn't incorrectly format an existing instance somewhere else in the doc(s).

**Comment by alfarok** [comment]
I started working on formatting all the existing docs that contain RPC components, and am manually verifying the rendering of the components after the formatting is applied.  I am going to knock this out so we can confidently run prettier without having to double check existing work.  I am finding in many cases that simply wrapping the RPC in the conditional rendering brackets solves most formatting related issues.

https://github.com/w3f/polkadot-wiki/pull/3732/files

**Comment by alfarok** [comment]
Addressed w/ #3790, #3732, and #3794
